src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -162,19 +162,37 @@ void ShenandoahScanRemembered::process_clusters(size_t first_cluster, size_t cou
       // [left, right) is a maximal right-open interval of dirty cards
       HeapWord* left = _rs->addr_for_card_index(dirty_l);        // inclusive
       HeapWord* right = _rs->addr_for_card_index(dirty_r + 1);   // exclusive
+      if (end_addr <= left) {
+        // The range of addresses to be scanned is empty
+        continue;
+      }
       // Clip right to end_addr established above (still exclusive)
       right = MIN2(right, end_addr);
       assert(right <= region->top() && end_addr <= region->top(), "Busted bounds");
       const MemRegion mr(left, right);
 
-      // NOTE: We'll not call block_start() repeatedly
-      // on a very large object if its head card is dirty. If not,
-      // (i.e. the head card is clean) we'll call it each time we
-      // process a new dirty range on the object. This is always
-      // the case for large object arrays, which are typically more
+      // NOTE: We'll not call first_object_start() repeatedly
+      // on a very large object, i.e. one spanning multiple cards,
+      // if its head card is dirty. If not, (i.e. its head card is clean)
+      // we'll call it each time we process a new dirty range on the object.
+      // This is always the case for large object arrays, which are typically more
```

Comment:
Instead of `This is always the case ...` may be we can say `The latter is aways the case ...`?

(Mea culpa for the old comment.)